### PR TITLE
Search: Add some dummy chart data & basic chart setup

### PR DESCRIPTION
**Pull request by annacmc**
Currently, this setup is how the standalone record meter app did work. However, now it's not loading and giving the console error `TypeError: Chart is not a constructor` when trying to use in wp-admin.

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+const Chart = React.lazy( () => import( 'chart.js/auto' ) );
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:5]: This was the culprit for me. When I swap the lazy-loading for a direct import, the chart works:

`import Chart from 'chart.js/auto';`

Screen Shot 2022-02-22 at 15 12 30

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
+/**
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:1]: We should rename this file to `bar-chart.jsx` to match the naming convention.

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			this.setState( { legendItems: this.myChart.legend.legendItems } );
+		}
+	}
+
+	render() {
+		if ( this.props.isValid === false ) {
+			return null;
+		}
+		return (
+			<div className="barChart">
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:84]: Our CSS class convention here is `bar-chart`.

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+				<div className="chartContainer">
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:85]: Our CSS class convention here is `bar-chart__container`.

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+						<p>Text alternative for this canvas graphic is in the data table below.</p>
+						<table
+							border="0"
+							cellPadding="5"
+							summary="This is the text alternative for the canvas graphic."
+						>
+							<caption>Records Indexed for Jetpack Search</caption>
+							<tbody>
+								<tr>
+									<th scope="col">Post Type</th>
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:96]: We need to translate these strings using something like `__( ' Post Type', 'jetpack-search-pkg' )`.

File: projects/packages/search/src/dashboard/components/record-meter/lib/recordInfo.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
+/**
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/recordInfo.js:1]: We should rename this file to record-info.js to match the naming convention.

File: projects/packages/search/src/dashboard/components/record-meter/lib/createData.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
+/**
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/createData.js:1]: We should rename this file to create-data.js to match the naming convention.

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx (hunk 1 above)
Comment (annacmc) [reply projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:5]: Oh, nice find! 👍  thank you! When I was chatting with @kangzj about this, he suggested that I'd need to use lazy load instead. However, perhaps I misunderstood that conversation. Is there any particular reason Jasper (or Chris!) that a direct import is a bad idea here?

Comment (kangzj) [reply projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:5]: `chartjs` is pretty big (about 200k after minified, dashboard alone is already 180k+), and it's a dedicated package, so it's a good idea to load it dynamically i.e. lazy loading, which we are using for some libs for Instant Search. However i think this is not a must. As to the issue, I guess, there is some initialization issue when doing lazyloading, which would need some sleuthing. 

Would 400k seem a sensible size? @bluefuton @jsnmoon

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx (hunk 3 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:84]: And a prefix might better, to avoid CSS conflicts, for example `jps_dashboard_bar_chart`

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx (hunk 1 above)
Comment (jsnmoon) [reply projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:5]: A quick summary of the situation:

- Jasper recommended we lazy load `chart.js`, which is indeed [nearly 200 kB minified](https://bundlephobia.com/package/chart.js@3.7.1) and dwarfs the dashboard bundle. Lazy loading large chunks of code can be beneficial, especially if we need to prioritize quick time-to-render (e.g. `/` on WordPress.com); if we wanted to ensure that our customers could quickly toggle Search and Instant Search, it'd be a good fit here.

- The current implementation wraps a dynamic `chart.js` import with a `React.lazy` invocation. This pattern is meant to lazily load other React components and does not work with importing general libraries (which we are doing here). If we were, however, we'd also need to wrap the use of such a component with a `Suspense` component as shown in the [docs](https://reactjs.org/docs/code-splitting.html#reactlazy).

To put it simply: lazy loading (aka code splitting) takes a bit more effort to have function correctly in React rendering. Fun fact, [suspense usage is currently discouraged in Preact](https://preactjs.com/guide/v10/switching-to-preact/#suspense-experimental) due to bugs. 🙃 

---

My recommendation here is to import `chart.js` as normal and push toward landing this code into production. Afterward, I recommend we invest time figuring out how to replace our `chart.js` usage with native SVG manipulations. (cc @bluefuton)

File: projects/packages/search/src/dashboard/components/record-meter/barchart.jsx (hunk 3 above)
Comment (bluefuton) [reply projects/packages/search/src/dashboard/components/record-meter/barchart.jsx:84]: Good call. Should be dashes to separate the prefix though :)

The main dashboard uses `jp-search-dashboard-page`.

Maybe we could use `jp-search-record-meter-bar-chart`?
